service/service.go

```diff
@@ -158,6 +158,34 @@ func (s *Service) CreateArtifactType(ctx context.Context, req *api.CreateArtifac
 	}, nil
 }
 
+// UpdateArtifactType update specified artifact type.
+// Currently, this function is used to add new fields to artifact type.
+// Deleting field is not supported and all fields must be matched.
+func (s *Service) UpdateArtifactType(ctx context.Context, req *api.UpdateArtifactTypeRequest) (*api.UpdateArtifactTypeResponse, error) {
+	var putTypeOptions mlmetadata.PutTypeOptions
+
+	if req.ArtifactType == nil {
+		return nil, errors.New("no ArtifactType specified")
+	}
+
+	putTypeOptions.AllFieldsMustMatch = true
+	putTypeOptions.CanAddFields = true
+
+	_, err := s.store.PutArtifactType(req.ArtifactType, &putTypeOptions)
```

Comment:
It is more readable if you can use struct literal:
```
s.store.PutArtifactType(req.ArtifactType, &putTypeOptions {
   AllFieldsMustMatch : true,
   CanAddFields: true,
})
```

Reply:
OK. No problem.